cmd/deploy.go

```diff
@@ -238,34 +237,49 @@ func autofillApimodel(dc *deployCmd) error {
 		}
 	}
 
-	if dc.dnsPrefix != "" && dc.containerService.Properties.MasterProfile.DNSPrefix != "" {
-		return fmt.Errorf("invalid configuration: the apimodel masterProfile.dnsPrefix and --dns-prefix were both specified")
-	}
-	if dc.containerService.Properties.MasterProfile.DNSPrefix == "" {
-		if dc.dnsPrefix == "" {
-			return fmt.Errorf("apimodel: missing masterProfile.dnsPrefix and --dns-prefix was not specified")
+	switch {
+	case dc.containerService.Properties.MasterProfile != nil:
```

Comment:
I feel like we're changing behavior here and I'm not sure of the knock-on effects.  Presumably, since there was never a check for MasterProfile != nil above this logic was always being run and these fields were always being set.  Now they will not be. Looks like you found a couple of cases just in this method.